### PR TITLE
Fixes #21009: add react scaffolding.

### PR DESCRIPTION
**Pull request by waldenraines**
Add basic react scaffolding to get us up and running with react
in katello.

http://projects.theforeman.org/issues/21009

### CODE REVIEW

File: lib/katello/plugin.rb [hunk 1]
```diff
@@ -30,13 +30,22 @@
...
+           :engine => Katello::Engine,
+           :turbolinks => false
+    else
+      menu :top_menu,
+           :redhat_provider,
+           :caption => N_('Red Hat Repositories'),
+           :url_hash => {:controller => 'katello/providers', :action => 'redhat_provider'},
+           :engine => Katello::Engine,
+           :turbolinks => false
+    end
```
Comment (waldenraines) [inline lib/katello/plugin.rb]: There may be a better way to do this but my rubyfu isn't the best.

**Comment by waldenraines** [comment]
Note that you can currently only see this at /redhat_repositories.

The menu item is updated if you turn on the experimental UI via Settings -> Content -> Show Experimental UI (requires a foreman restart).

**Comment by waldenraines** [comment]
Requires https://github.com/theforeman/foreman/pull/4733 and specifically the diff in https://github.com/theforeman/foreman/pull/4733#issuecomment-332848660.

**Comment by waldenraines** [comment]
[test]

File: app/views/katello/layouts/react.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<% content_for(:javascripts) do %>
+  <%= javascript_include_tag *webpack_asset_paths('katello', :extension => 'js'), "data-turbolinks-track" => true %>
```
Comment (tstrachota) [inline app/views/katello/layouts/react.html.erb]: I wonder if we should add this path also to `katello.html.erb` layout to unlock using new webpack stuff in angular code.

**Comment by tstrachota** [comment]
Works fine from what I tested. The required PR in the Foreman repo has been merged so it's no longer blocked.

It would be good to add `node_modules` to gitignore.

**Comment by waldenraines** [comment]
@tstrachota thanks for the review and apologies that I am now closing this unmerged in favor of an approach that @danseethaler came up with to be PR'd soon.

**Pull request closed without merging**